SYCL/Plugin/interop-cuda-experimental.cpp

```diff
@@ -0,0 +1,82 @@
+// REQUIRES: cuda
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -lcuda %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#define SYCL_EXT_ONEAPI_BACKEND_CUDA_EXPERIMENTAL 1
+#include <sycl/ext/oneapi/experimental/backend/cuda.hpp>
+#include <sycl/sycl.hpp>
+
+#include <cuda.h>
+
+#include <assert.h>
+
+void cuda_check(CUresult error) { assert(error == CUDA_SUCCESS); }
+
+template <typename refT, typename T> void check_type(T var) {
+  bool is_same = std::is_same_v<decltype(var), refT>;
+  assert(is_same);
+}
+
+#define CUDA_CHECK(error) cuda_check(error)
+
+bool check_queue(sycl::queue &Q) {
+  constexpr size_t vec_size = 5;
+  double A_Data[vec_size] = {4.0};
+  double B_Data[vec_size] = {-3.0};
+  double C_Data[vec_size] = {0.0};
+
+  sycl::buffer<double, 1> A_buff(A_Data, sycl::range<1>(vec_size));
+  sycl::buffer<double, 1> B_buff(B_Data, sycl::range<1>(vec_size));
+  sycl::buffer<double, 1> C_buff(C_Data, sycl::range<1>(vec_size));
+
+  Q.submit([&](sycl::handler &cgh) {
+     auto A_acc = A_buff.get_access<sycl::access::mode::read>(cgh);
+     auto B_acc = B_buff.get_access<sycl::access::mode::read>(cgh);
+     auto C_acc = C_buff.get_access<sycl::access::mode::write>(cgh);
+     cgh.parallel_for(sycl::range<1>{vec_size}, [=](sycl::id<1> idx) {
+       C_acc[idx] = A_acc[idx] + B_acc[idx];
+     });
+   }).wait();
+
+  sycl::host_accessor C_acc(C_buff);
+  return C_acc[0] == 1;
+}
+
+int main() {
+  sycl::queue Q;
```

Comment:
Is this used anywhere? If not, then maybe it would be interesting to do the `get_native` -> `make_*` -> `check` with `Q` and the associated context and device.

Additionally, if that is added it would be good to check that `Q` still works afterwards. Assuming the CUDA backend doesn't take ownership, which I don't remember whether or not it does.

Reply:
Thats a great idea. I have added an additional check by creating a queue with `Q`'s context and then checking both the new queue and the original work.